pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -184,36 +213,213 @@ func (r *MachineSyncReconciler) reconcileCAPIMachinetoMAPIMachine(ctx context.Co
+
+	if capiMachine == nil {
+		if err := r.Create(ctx, newCAPIMachine); err != nil {
+			logger.Error(err, "Failed to create CAPI machine")
+
+			createErr := fmt.Errorf("failed to create CAPI machine: %w", err)
+			if condErr := r.applySynchronizedConditionWithPatch(
+				ctx, mapiMachine, corev1.ConditionFalse, reasonFailedToCreateCAPIMachine, createErr.Error(), nil); condErr != nil {
+				return ctrl.Result{}, utilerrors.NewAggregate([]error{createErr, condErr})
+			}
+
+			return ctrl.Result{}, createErr
+		}
+
+		logger.Info("Successfully created CAPI machine")
+
+		return ctrl.Result{}, nil
+	}
+
+	capiMachinesDiff := compareCAPIMachines(capiMachine, newCAPIMachine)
+
+	if len(capiMachinesDiff) == 0 {
+		logger.Info("No changes detected in CAPI machine")
+		return ctrl.Result{}, nil
+	}
+
+	logger.Info("Changes detected, updating CAPI machine", "diff", capiMachinesDiff)
+
+	if err := r.Update(ctx, newCAPIMachine); err != nil {
+		logger.Error(err, "Failed to update CAPI machine")
+
+		updateErr := fmt.Errorf("failed to update CAPI machine: %w", err)
+
+		if condErr := r.applySynchronizedConditionWithPatch(ctx, mapiMachine, corev1.ConditionFalse, reasonFailedToUpdateCAPIMachine, updateErr.Error(), nil); condErr != nil {
+			return ctrl.Result{}, utilerrors.NewAggregate([]error{updateErr, condErr})
+		}
+
+		return ctrl.Result{}, updateErr
+	}
+
+	logger.Info("Successfully updated CAPI machine")
+
 	return ctrl.Result{}, nil
 }
 
-// getInfraMachineFromProvider returns the correct InfraMachine implementation
+// initInfraMachineAndInfraClusterFromProvider returns the correct InfraMachine and InfraCluster implementation
 // for a given provider.
 //
 // As we implement other cloud providers, we'll need to update this list.
-func getInfraMachineFromProvider(platform configv1.PlatformType) (client.Object, error) {
+func initInfraMachineAndInfraClusterFromProvider(platform configv1.PlatformType) (client.Object, client.Object, error) {
 	switch platform {
 	case configv1.AWSPlatformType:
-		return &capav1beta2.AWSMachine{}, nil
+		return &capav1beta2.AWSMachine{}, &capav1beta2.AWSCluster{}, nil
 	case configv1.PowerVSPlatformType:
-		return &capibmv1.IBMPowerVSMachine{}, nil
+		return &capibmv1.IBMPowerVSMachine{}, &capibmv1.IBMPowerVSCluster{}, nil
 	default:
-		return nil, fmt.Errorf("%w: %s", errPlatformNotSupported, platform)
+		return nil, nil, fmt.Errorf("%w: %s", errPlatformNotSupported, platform)
 	}
 }
 
 // shouldMirrorCAPIMachineToMAPIMachine takes a CAPI machine and determines if there should
 // be a MAPI mirror, it returns true only if:
 //
-// 1. The CAPI Machine is owned by a CAPI MachineSet,
-// 2. That owning CAPI MachineSet has a MAPI MachineSet Mirror.
+// 1. The CAPI Machine is owned by a CAPI Machine,
+// 2. That owning CAPI Machine has a MAPI Machine Mirror.
 func (r *MachineSyncReconciler) shouldMirrorCAPIMachineToMAPIMachine(ctx context.Context, logger logr.Logger, machine *capiv1beta1.Machine) (bool, error) {
```

Comment:
Looks like every word MachineSet was changed to Machine in this function.

Reply:
Fixed, thanks.

Reply:
This is still not resolved. It's not just this comment, but every occurrence of the word MachineSet in this function.